homeassistant/components/keba/binary_sensor.py

```diff
@@ -0,0 +1,147 @@
+
+SENSORS = {
+    'Online': ['Wallbox', 'connectifity'],
+    'Plug': ['Plug', 'plug'],
+    'State': ['Charging state', 'power'],
+    'Tmo FS': ['Failsafe Mode', 'safety'],
+}
+
+
+async def async_setup_platform(hass, config,
+                               async_add_entities, discovery_info=None):
+    """Set up the KEBA charging station platform."""
+    _LOGGER.debug("Initializing KEBA charging station binary sensors")
+    sensors = []
+    keba = hass.data[DOMAIN]
+
+    for key, value in sorted(SENSORS.items()):
+        sensors.append(KebaBinarrySensor(key, keba, value[0], value[1]))
+
+    async_add_entities(sensors)
+
+
+class KebaBinarrySensor(BinarySensorDevice):
+    """Representation of a binary sensor of a KEBA charging station."""
+
+    def __init__(self, key, keba, sensor_name, device_class):
+        """Initialize the KEBA Sensor."""
+        self._key = key
+        self._keba = keba
+        self._name = sensor_name
+        self._device_class = device_class
+        self._state = None
+        self._attributes = {}
+
+    @property
+    def should_poll(self):
+        """Return False.
+
+        Data update is triggered from Keba entitiy.
+        """
+        return False
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of the binary sensor."""
+        return "keba_" + self._name
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return "keba_" + self._name
+
+    @property
+    def device_class(self):
+        """Return the class of this sensor."""
+        return self._device_class
+
+    @property
+    def is_on(self):
+        """Return true if sensor is on."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the binary sensor."""
+        return self._attributes.items()
+
+    async def async_update(self):
+        """Get latest cached states from the device."""
+        if self._key == 'Online':
+            self._state = self._keba.get_value(self._key)
+            self._attributes['Product'] = self._keba.get_value('Product')
+            self._attributes['Serial'] = self._keba.get_value('Serial')
+            self._attributes['Firmware'] = self._keba.get_value('Firmware')
+
+            secs = self._keba.get_value('Sec')
+
+            text = ""
+            if secs is not None:
+                if secs < 60:
+                    text = str(int(secs)) + " secs"
+                elif secs < 3600:
+                    text = str(int(secs / 60)) + " mins"
+                elif secs < 86400:
+                    text = str(int(secs / 3600)) + " hours"
+                elif secs < 31536000:
+                    text = str(int(secs / 86400)) + " days"
+                else:
+                    text = str(int(secs / 31536000)) + " years, " + \
+                           str(int((secs % 31536000) / 86400)) + " days"
+                self._attributes['Device uptime'] = text
```

Comment:
Personally I would add a method to the backend library to get "prettified" version for a value (`get_value('Sec', pretty=True)`), and do that conversion there. Would timedelta not be pretty enough for times though?
```
>>> str(timedelta(seconds=36))
'0:00:36'
>>> str(timedelta(seconds=360))
'0:06:00'
>>> str(timedelta(seconds=3601))
'1:00:01'
>>> str(timedelta(seconds=36001))
'10:00:01'
>>> str(timedelta(seconds=360001))
'4 days, 4:00:01'
>>> str(timedelta(seconds=3600001))
'41 days, 16:00:01'
>>> str(timedelta(seconds=36000001))
'416 days, 16:00:01'
```

Reply:
Thanks for this hint, I added it in the library 0.1.1